### PR TITLE
[e2e] allow sandbox

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Supports running e2e tests out of `agent-sandbox` in local runs.  

### Motivation

Some teams out of agent are interested in using the new e2e framework, but they cannot use `agent-sandbox` as it is restricted to agent folks. This PR allows to override the default `agent-sandbox` with anything else. 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/runner/parameters/fixtures/config_with_stackparams.yaml [hunk 1]
```diff
@@ -1,12 +1,14 @@
 # Environments configuration, validated and passed to Pulumi
 configParams:
   aws:
+    # aws account where resources will be created on
+    account: "kiki"
```
Comment (ogaca-dd) [inline test/new-e2e/runner/parameters/fixtures/config_with_stackparams.yaml:5]: Is this name correct?

**Review by ogaca-dd — APPROVED** [review]

File: test/new-e2e/runner/parameters/fixtures/config_with_stackparams.yaml (hunk 1 above)
Comment (pducolin) [reply test/new-e2e/runner/parameters/fixtures/config_with_stackparams.yaml:5]: This is used in a test, might rename `fixture` to `testfixture` to be more explicit. The logic here is different from the one in `test-infra`, as I don't check for the validity of the account. The validation is at `inv setup` level in `test-infra`, once here I expect the local config to be valid, so I accept any value.

File: test/new-e2e/runner/local_profile.go [hunk 2]
```diff
@@ -26,17 +26,21 @@ func NewLocalProfile() (Profile, error) {
...
 	}
 
+	awsAccount := "agent-sandbox"
 	var store parameters.Store
 	if configPath != "" {
 		configFileValueStore, err := parameters.NewConfigFileValueStore(configPath)
 		if err != nil {
 			return nil, fmt.Errorf("error when reading the config file %v: %v", configPath, err)
 		}
+		if configFileValueStore.Config.ConfigParams.AWS.Account != "" {
```
Comment (vboulineau) [inline test/new-e2e/runner/local_profile.go:36]: In my opinion this implementation is somewhat hackish. When reading parameters, you're not suppose to know the backing store.

Here your goal is to set environments from a variable. What should be done is to add a parameter in store with `env` that is read and defaulted to `aws/agent-sandbox` if empty.
It should like any param and not limited to the file.

Comment (pducolin) [reply test/new-e2e/runner/local_profile.go:36]: Will move default to `store_config_file`, makes sense to move logic there

Comment (pducolin) [reply test/new-e2e/runner/local_profile.go:36]: One additional question is if it makes sense to have such a complex setup for local config. Wouldn't it be enough to use the local file ? How many people use env variables ?

Comment (pducolin) [reply test/new-e2e/runner/local_profile.go:36]: Side note: ci_profile will require refactoring too, if we want to have a single store in the end and get variables from it.

Comment (pducolin) [reply test/new-e2e/runner/local_profile.go:36]: Looking at code, moving this to a store parameter and load it in the common profile code is too risky as `ci_profile` knows about its backing store, plus reads env variables straight from the CI, without the store. I suggest we postpone the refactoring to a follow up PR, let me know if this is blocking for you

File: test/new-e2e/runner/ci_profile.go [hunk 3]
```diff
@@ -15,7 +15,7 @@ import (
 
 const defaultCISecretPrefix = "ci.datadog-agent."
 
-var defaultCIEnvs = []string{"aws/agent-qa"}
+var defaultCIEnvs = "aws/agent-qa"
```
Comment (vboulineau) [inline test/new-e2e/runner/ci_profile.go:18]: s/var/const to be bundled with above `const`, name to be aligned with the one in `local_profile.go`.

File: test/new-e2e/runner/parameters/store_config_file.go [hunk 4]
```diff
@@ -46,7 +47,7 @@ type Agent struct {
 var _ valueStore = &configFileValueStore{}
 
 type configFileValueStore struct {
-	config          Config
+	Config          Config
```
Comment (vboulineau) [inline test/new-e2e/runner/parameters/store_config_file.go:50]: Config should now go back to private.

File: test/new-e2e/runner/ci_profile.go [hunk 5]
```diff
@@ -13,9 +13,10 @@ import (
 	"github.com/DataDog/datadog-agent/test/new-e2e/runner/parameters"
 )
 
-const defaultCISecretPrefix = "ci.datadog-agent."
-
-var defaultCIEnvs = []string{"aws/agent-qa"}
+const (
+	defaultCISecretPrefix = "ci.datadog-agent."
+	defaultCIEnv          = "aws/agent-qa"
```
Comment (vboulineau) [inline test/new-e2e/runner/ci_profile.go:18]:
```suggestion
	defaultCIEnvironments          = "aws/agent-qa"
```

File: test/new-e2e/runner/ci_profile.go [hunk 6]
```diff
@@ -51,12 +52,20 @@ func NewCIProfile() (Profile, error) {
...
-	ciEnvironmentsStr := os.Getenv("CI_ENV_NAMES")
-	if len(ciEnvironmentsStr) > 0 {
-		ciEnvironments = strings.Split(ciEnvironmentsStr, " ")
+	// get environments from store
+	environmentsStr, err := store.GetWithDefault(parameters.Environments, defaultCIEnv)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO can be removed using E2E_ENVIRONMENTS variable
```
Comment (vboulineau) [inline test/new-e2e/runner/ci_profile.go:61]:
```suggestion
	// TODO can be removed using E2E_ENV variable
```

File: test/new-e2e/runner/ci_profile.go (hunk 5 above)
Comment (pducolin) [reply test/new-e2e/runner/ci_profile.go:18]: Another developer might say that `defaultCIEnvironments` is a string, thus a single environment.

Comment (pducolin) [reply test/new-e2e/runner/ci_profile.go:18]: Will commit, no strong opinion

**Review by vboulineau — APPROVED** [review]

**Pull request merged**